oktapam/client/secret_folder.go

```diff
@@ -0,0 +1,309 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/atko-pam/pam-sdk-go/client/pam"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/logging"
+	"github.com/tomnomnom/linkheader"
+)
+
+const secretFolderSeparator = "/"
+
+// SecretFolder represents a secret folder entity.  We use this instead of the PAM SDK response since
+// there are multiple API response types which are used for secret folders and this provides a common
+// structure that can be used
+type SecretFolder struct {
```

Comment:
I didn't look at the spec, but can't we use same schema in the spec too ?

Reply:
I think there are some slight differences between the responses.  It's possible that we can merge this into a single spec, but I think that's outside the scope of this PR.